lib/Map/UserDrawing.js

```diff
@@ -0,0 +1,400 @@
+     */
+    var svgDataDeclare = "data:image/svg+xml,";
+    var svgPrefix = '<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" width="20px" height="20px" xml:space="preserve">';
+    var svgCircle = '<circle cx="10" cy="10" r="5" stroke="rgb(0,170,215)" stroke-width="4" fill="white" /> ';
+    var svgSuffix = "</svg>";
+    var svgString = svgPrefix + svgCircle + svgSuffix;
+
+    // create the cesium entity
+    this.svgPoint = svgDataDeclare + svgString;
+};
+
+/**
+ * Start interpreting user clicks as placing or removing points.
+ */
+UserDrawing.prototype.enterDrawMode = function() {
+
+    // If we have finished a polygon, don't allow more points to be drawn. In future, perhaps support multiple polygons.
+    if (this.inDrawMode || this.closeLoop) {
+        // Do nothing
+        return;
+    }
+
+    this.inDrawMode = true;
+    document.getElementById("cesiumContainer").setAttribute("style", "cursor: crosshair");
+
+    // Cancel any feature picking already in progress.
+    this.terria.pickedFeatures = undefined;
+    var that = this;
+
+    // Line will show up once user has drawn some points. Vertices of line are user points.
+    this.otherEntities.entities.add({
+        name: 'Line',
+        polyline: {
+                    positions: new CallbackProperty(function(date, result) {
+                        var pos = that._getPointsForShape();
+                        if (that.closeLoop) {
+                            pos.push(pos[0]);
+                        }
+                        return pos;
+                    }, false),
+
+                    material : new PolylineGlowMaterialProperty({
+                        color : new Color(0.0, 0.0, 0.0, 0.1),
+                        glowPower : 0.25
+                    }),
+                    width: 20
+                  }
+    });
+    this.terria.dataSources.add(this.pointEntities);
+    this.terria.dataSources.add(this.otherEntities);
+
+    // Listen for user clicks on map
+    const pickPointMode = new MapInteractionMode({
+        message: this._getDialogMessage(),
+        buttonText: this._getButtonText(),
+        onCancel: function() {
+            that.terria.mapInteractionModeStack.pop();
+            that._cleanUp();
+        }
+    });
+    this.terria.mapInteractionModeStack.push(pickPointMode);
+
+    // Handle what happens when user picks a point
+    knockout.getObservable(pickPointMode, 'pickedFeatures').subscribe(function(pickedFeatures) {
+        when(pickedFeatures.allFeaturesAvailablePromise, function() {
+            if (defined(pickedFeatures.pickPosition)) {
+                var pickedPoint = pickedFeatures.pickPosition;
+                var firstPointEntity = new Entity({
+                    name: 'First Point',
+                    position: pickedPoint,
+                    billboard : {
+                        image : that.svgPoint,
```

Comment:
I thought Cesium had problems using SVGs in billboards in some browsers.  Did I make that up?